packages/cashier/src/containers/routes/binary-routes.tsx

```diff
@@ -18,10 +20,24 @@ const Loading = () => {
 };
 
 const BinaryRoutes = (props: TBinaryRoutesProps) => {
+    const { is_p2p_v2_enabled } = useFeatureFlags();
+    const [routesConfig, setRoutesConfig] = React.useState(getRoutesConfig());
+    useEffect(() => {
+        if (is_p2p_v2_enabled) {
+            const routes_replicate = routesConfig;
+            routes_replicate[0].routes?.push({
+                path: routes.cashier_p2p_v2,
+                getTitle: () => localize('Deriv P2P-V2'),
+                icon_component: 'IcDp2p',
+            });
+            setRoutesConfig(routes_replicate);
+        }
+    }, [is_p2p_v2_enabled, routesConfig]);
```

Comment:
@ali-hosseini-deriv, does this work?
```suggestion
            setRoutesConfig(prevRoutes => {
                const routes_replicate = prevRoutes;
                routes_replicate[0].routes?.push({
                    path: routes.cashier_p2p_v2,
                    getTitle: () => localize('Deriv P2P-V2'),
                    icon_component: 'IcDp2p',
                });
            });
            return routes_replicate;
        }
    }, [is_p2p_v2_enabled]);
```

Reply:
why it's needed to be removed?

Reply:
Because it's being updated inside the same `effect` where it's listed as a dependancy, these tend to cause infinite loops / re-renders, incase something is added in the future.